tests/integration/test_backup_restore_new/test.py

```diff
@@ -191,6 +191,52 @@ def test_incremental_backup():
     assert instance.query("SELECT count(), sum(x) FROM test.table2") == "102\t5081\n"
 
 
+def test_incremental_backup_overflow():
+    backup_name = new_backup_name()
+    incremental_backup_name = new_backup_name()
+    incremental_backup_name_raw = backup_id_counter
+
+    instance.query("CREATE DATABASE test")
+    instance.query(
+        "CREATE TABLE test.table(y String CODEC(NONE)) ENGINE=MergeTree ORDER BY tuple()"
+    )
+    # Create a column of 4GB+10K
+    instance.query(
+        "INSERT INTO test.table SELECT toString(repeat('A', 1024)) FROM numbers((4*1024*1024)+10)"
+    )
+    # Force one part
+    instance.query("OPTIMIZE TABLE test.table FINAL")
+
+    # ensure that the column's size on disk is indeed greater then 4GB
+    assert (
+        int(
+            instance.query(
+                "SELECT bytes_on_disk FROM system.parts_columns WHERE active AND database = 'test' AND table = 'table' AND column = 'y'"
+            )
+        )
+        > 4 * 1024 * 1024 * 1024
+    )
+
+    instance.query(f"BACKUP TABLE test.table TO {backup_name}")
+    instance.query(
+        f"BACKUP TABLE test.table TO {incremental_backup_name} SETTINGS base_backup = {backup_name}"
+    )
+
+    # And now check that incremental backup does not have any files
+    assert (
+        int(
+            instance.exec_in_container(
+                [
+                    "bash",
+                    "-c",
+                    f"find /backups/{incremental_backup_name_raw} -type f ! -name .backup | wc -l",
```

Comment:
There is the function `get_path_to_backup()` defined in this `test.py` which allows to do such checking easier (without `exec_in_container`)

Reply:
Indeed, thanks!